components/views/dropdown-menu-view.js

```diff
@@ -0,0 +1,231 @@
+define([
```

Comment:
That's a good point. We need to work out which JS doc generation tool we're going to use, and where it would look for class level doc. I see that @AlasdairSwan has an excellent doc comment below (maybe added in response to your comment) which maybe is the better place to have it.